### PR TITLE
ui: Display unknown and failing alerts

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

prom

### CODE REVIEW

File: web/ui/react-app/src/pages/alerts/AlertContents.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ export interface Rule {
   alerts: Alert[];
   annotations: Record<string, string>;
   duration: number;
-  health: string;
+  health: RuleState;
```
Comment (juliusv) [inline web/ui/react-app/src/pages/alerts/AlertContents.tsx]: I'd prefer if we could only allow the actually valid values for `health` and `state` here. Maybe just get rid of the `export type RuleState = keyof RuleStatus<any>;` and do a direct `health: 'firing' | 'pending' | 'inactive'` here and `state: 'unknown' | 'err' | 'ok'` below? The alternative would be to have two separate enums / interfaces at the beginning and then joining them into one later on somehow.

**Comment by juliusv** [comment]
Sorry btw., I totally didn't see this. Best to @ me on UI PRs :)

File: web/ui/react-app/src/pages/alerts/AlertContents.tsx [hunk 2]
```diff
@@ -49,6 +53,9 @@ const AlertsContent: FC<AlertsProps> = ({ groups = [], statsCount }) => {
     firing: true,
     pending: true,
     inactive: true,
+    unknown: false,
+    err: false,
+    ok: false,
```
Comment (juliusv) [inline web/ui/react-app/src/pages/alerts/AlertContents.tsx]: Just so I understand correctly, you are introducing three new rule states here, and we have show/hide buttons for the original inactive/pending/firing, but not for the new ones, and it seems the new ones are always shown, with no option of filtering them. I also don't know if we should be adding three new buttons, but curious what your thinking is here.

**Comment by juliusv** [comment]
@roidelapluie Heya, just want to make sure, what's the background for closing this, was it intentional?

**Comment by roidelapluie** [comment]
Yes that means that I stop working on it at the moment and that I think my first approach is not good enough :)

**Comment by juliusv** [comment]
Ah ok, thanks for the clarification!

**Pull request closed without merging**